lib/linter/linter.js

```diff
@@ -456,10 +460,7 @@ function normalizeEcmaVersion(parser, ecmaVersion) {
     return ecmaVersion >= 2015 ? ecmaVersion - 2009 : ecmaVersion;
 }
 
-const eslintEnvPatterns = [
-    /\/\*\s*eslint-env\s(.+?)\*\//gsu, // block comments
-    /\/\/[^\S\r\n\u2028\u2029]*eslint-env[^\S\r\n\u2028\u2029](.+)/gu // line comments
-];
+const eslintEnvPattern = /\/\*\s*eslint-env\s([^*]+?)\*\//gsu;
```

Comment:
This will change behavior, for example:

```js
/* eslint no-undef: error */

/* eslint-env *, browser */

window
```

This comment doesn't match with the new regex.

The old regex finds this as a directive (later processing discards `*` as an unexisting environment name), and `browser` is applied - [Demo](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXVuZGVmOiBlcnJvciAqL1xuXG4vKiBlc2xpbnQtZW52ICosIGJyb3dzZXIgKi9cblxud2luZG93Iiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjoxMiwic291cmNlVHlwZSI6InNjcmlwdCIsImVjbWFGZWF0dXJlcyI6e319LCJydWxlcyI6e30sImVudiI6e319fQ==).

Reply:
it is to fix the codeql warning: https://github.com/eslint/eslint/pull/14973/checks?check_run_id=3411736426 (the old regex can be very slow in some edge cases.)

TBH, `/* eslint-env *, browser */` is not a valid directive, so its behavior is somewhat "undefined" - users should not rely on this kind of functionality.

Reply:
Would this fix the warning:

```js
const eslintEnvPattern = /\/\*\s*eslint-env\s(.+?(?:\*\/|$))/gsu;
```

That way, it would stop searching whenever a `/* eslint-env` doesn't end with `*/` (in my understanding, that was the problem), but we'll have to manually check if `*/` was captured and remove it.

Reply:
sounds good! 👍

Reply:
This might look better:

```js
const eslintEnvPattern = /\/\*\s*eslint-env\s(.+?)(?:\*\/|$)/gsu;
```

so that we don't have to slice off `*/` from `match[1]`, and we could check if `match[0]` ends with `*/`.

Reply:
awesome! I've tested it, and it's working well!

Reply:
Do you think we need to add the test?  I prefer no, as it is not documented, and not recommended.

Reply:
I was also able to reproduce the reported codeql problem with the old regex, but not with the new one. Let's just see if codeql will be able to figure out that.

Reply:
Ah, it's failing again, but I think it's a false positive now.

Reply:
> Do you think we need to add the test? I prefer no, as it is not documented, and not recommended.

It might be good for testing how our comment parsing works, but I think it isn't necessary.